pkg/cluster/nodes/node.go

```diff
@@ -224,3 +225,33 @@ func (n *Node) WriteKubeConfig(dest string) error {
 
 	return ioutil.WriteFile(dest, buff.Bytes(), 0600)
 }
+
+// WaitForControlPlane uses kubectl inside the "node" container to check if
+// the control plane node status is "Ready". The control plane node is found by
+// looking for the node containing the substring "control-plane" in its name.
+func (n *Node) WaitForControlPlane(until time.Time) bool {
```

Comment:
we can use a selector instead, kubeadm applies the label `node-role.kubernetes.io/master: ""`

Reply:
with regard to a kubectl based check, i think the ready state on the node object is a safer bet, because the taints and labels are a `kubeadm init phase` that can be omitted using `kubeadm init --skip-phases=mark-control-plane` in 1.13.

on the other hand, i'm pretty sure i've seen a case where a node can be ready while some pods are failing, so from my experience watching for an expected list of pods is the only truth of a healthy node.

Reply:
we're not waiting to see if it's healthy, we're waiting for scheduling to likely not fail for test pods in https://testgrid.k8s.io/conformance-kind#kind,%20master%20(dev)%20%5Bnon-serial%5D, 

I don't think kind should build it's own healthchecking yet. but logically workloads cannot schedule if the nodes are not in the ready state.

Reply:
for workloads this is OK.
for things like `join --experimental-control-plane` it can fail if the existing etcd cluster is not ready to join yet another etcd node. there are some tricky timeouts around HA, but we shall see.